### PR TITLE
PullRequest for Issue2342: code refactor to rename some bioxas classes to CLS classes so that PGM can use them

### PR DESCRIPTION
**Pull request by XuetaoLiu-CLS**
#2342 
Other than #2127, this issue will only change the BioXAS classes to the CLS ones, so that PGM can use it. #2127 will make further code refactor

### CODE REVIEW

**Comment by XuetaoLiu-CLS** [comment]
@helfrij can you review this ? thx

**Comment by XuetaoLiu-CLS** [comment]
@iainworkman As @helfrij recommended, you might be the good candidate to review this.

File: clsCommon.pri [hunk 1]
```diff
@@ -32,6 +39,16 @@ HEADERS *=\
 	source/dataman/info/CLSOceanOptics65000DetectorInfo.h \
 	source/dataman/info/CLSAmptekSDD123DetectorInfo.h \
 	source/dataman/info/CLSSynchronizedDwellTimeConfigurationInfo.h \
+	source/ui/CLS/CLSValueProgressLabel.h \
+	source/ui/CLS/CLSValueEditor.h \
+	source/ui/CLS/CLSControlEditor.h \
```
Comment (iainworkman) [inline clsCommon.pri]: Is it time to think about doing the full replace of AMControlEditor with your better version (as a separate issue) @helfrij?

File: source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #include "beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.h"
 
 BioXASSSRLMonochromatorRegionControlEditor::BioXASSSRLMonochromatorRegionControlEditor(BioXASSSRLMonochromatorRegionControl *regionControl, QWidget *parent) :
-	BioXASControlEditor(regionControl, parent)
+	CLSControlEditor(regionControl, parent)
 {
 
 }
```
Comment (iainworkman) [inline source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp]: Line 21 in this file still refers to BIOXASXControlEditor::setControl(newControl);

Should be CLSControlEditor, I believe.

Comment (helfrij) [inline source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp]: If it were up to me, I would prefer to handle changes to BioXASControlEditor as a separate issue. I'm not sure it's ready for use throughout the CLS yet--I'm still doing some development work on it.

File: clsCommon.pri (hunk 1 above)
Comment (XuetaoLiu-CLS) [inline clsCommon.pri]: @helfrij @iainworkman 

There is another issue #2127, which might handle this.

File: source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp (hunk 2 above)
Comment (XuetaoLiu-CLS) [inline source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp]: @iainworkman good catch. I made the changes yesterday last minute before I left and forgot to save the cpp file :).

**Comment by iainworkman** [comment]
Not sure if this has come through with this pull request (it doesn't seem so) but I'm getting a compile error on line 73 of BioXASSideAppController where it can't find onGeDetectorConnectedChanged()

**Comment by iainworkman** [comment]
Other than that, looks good.
Time Estimate: 1 hour

**Comment by XuetaoLiu-CLS** [comment]
@iainworkman Great catch. I guess I didn't get the compile error because I didn't make a clean build.

**Comment by XuetaoLiu-CLS** [comment]
@dretrex can you take a look at this PR and merge it in? Then we can review the PGM PR #2209 . Thx

File: source/beamline/CLS/CLSSSRLMonochromatorMaskState.cpp [hunk 3]
```diff
@@ -1,8 +1,9 @@
-#include "BioXASSSRLMonochromatorMaskState.h"
+#include "CLSSSRLMonochromatorMaskState.h"
+
 #include "actions3/AMActionSupport.h"
 #include "actions3/AMListAction3.h"
```
Comment (helfrij) [inline source/beamline/CLS/CLSSSRLMonochromatorMaskState.cpp]: Hey @XuetaoLiu-CLS, why was an SSRL mono class made into a CLS class here? Is it going to be used somewhere else?

File: source/beamline/CLS/CLSMirrorMaskState.h [hunk 4]
```diff
@@ -1,9 +1,9 @@
-#ifndef BIOXASM1MIRRORMASKSTATE_H
-#define BIOXASM1MIRRORMASKSTATE_H
+#ifndef CLSMIRRORMASKSTATE_H
+#define CLSMIRRORMASKSTATE_H
 
 #include "beamline/AMSingleEnumeratedControl.h"
 
-class BioXASM1MirrorMaskState : public AMSingleEnumeratedControl
+class CLSMirrorMaskState : public AMSingleEnumeratedControl
```
Comment (helfrij) [inline source/beamline/CLS/CLSMirrorMaskState.h]: Same question with the BioXAS mirror class here?

**Pull request merged**